utils/models_common.py

```diff
@@ -0,0 +1,44 @@
+from glob import glob
+import os
+from pathlib import Path
+import torch
+import cv2
+import torchvision
+from tqdm import tqdm
+import logging
+import psutil
+from torch.utils.tensorboard import SummaryWriter
+import numpy as np
+
+
+def np_to_tensor(x, device):
+    if device == 'cpu':
+        return torch.from_numpy(x).cpu()
+    else:
+        return torch.from_numpy(x).contigious().pin_memory().to(device=device, non_blocking=True)
+
+
+class Nop(object):
+    def nop(*args, **kw): pass
+
+    def __getattr__(self, _): return self.nop
+
+
+def train_(train_dataloader, eval_dataloader, caller):
```

Comment:
Ignore from here onwards, still trying to come up with a general enough train function